spec.md

```diff
@@ -1663,6 +1716,509 @@ message GetCapacityResponse {
+
+  // Secrets required by plugin to complete group snapshot deletion
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(csi_secret) = true];
+}
+
+message DeleteVolumeGroupSnapshotResponse {
+  // Intentionally empty.
+}
+```
+
+##### DeleteVolumeGroupSnapshot Errors
+
+If the plugin is unable to complete the DeleteVolumeGroupSnapshot call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the conditions defined below are encountered, the plugin MUST return the specified gRPC error code.
+The CO MUST implement the specified error recovery behavior when it encounters the gRPC error code.
+
+| Condition | gRPC Code | Description | Recovery Behavior |
+|-----------|-----------|-------------|-------------------|
+| Volume group snapshot in use | 9 FAILED_PRECONDITION | Indicates that the volume group snapshot corresponding to the specified `group_snapshot_id` could not be deleted because it is in use by another resource. | Caller SHOULD ensure that there are no other resources using the volume group snapshot, and then retry with exponential back off. |
+
+#### `ControllerGetVolumeGroupSnapshot`
+
+**ALPHA FEATURE**
+
+```protobuf
+message ControllerGetVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The ID of the group snapshot to fetch current group snapshot
+  // information for.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+
+  // Secrets required by plugin to complete
+  // ControllerGetVolumeGroupSnapshot request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(csi_secret) = true];
+}
+
+message ControllerGetVolumeGroupSnapshotResponse {
+  option (alpha_message) = true;
+
+  // This field is REQUIRED
+  VolumeGroupSnapshot group_snapshot = 1;
+}
+```
+
+##### ControllerGetVolumeGroupSnapshot Errors
+
+If the plugin is unable to complete the ControllerGetVolumeGroupSnapshot call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the conditions defined below are encountered, the plugin MUST return the specified gRPC error code.
+The CO MUST implement the specified error recovery behavior when it encounters the gRPC error code.
+
+| Condition | gRPC Code | Description | Recovery Behavior |
+|-----------|-----------|-------------|-------------------|
+| Volume group snapshot does not exist | 5 NOT_FOUND | Indicates that a volume group snapshot corresponding to the specified `volume_group_snapshot_id` does not exist. | Caller MUST verify that the `volume_group_snapshot_id` is correct and that the volume group snapshot is accessible and has not been deleted before retrying with exponential back off. |
+
+#### `ListVolumeGroupSnapshots`
+
+**ALPHA FEATURE**
+
+```protobuf
+message ListVolumeGroupSnapshotsRequest {
+  option (alpha_message) = true;
+
```

Comment:
Would you want a capability here to list only the VolumeGroupSnapshots from a particular VolumeGroup, by specifying an optional volume_group_id?

Reply:
Good point.  Added.

Reply:
Thanks.